### PR TITLE
apply suggestions from MSVS

### PR DESCRIPTION
**Pull request by Aprogiena**
This is what MSVS suggest to do in compiler messages (i.e. not errors, not warnings, but lowest level).

### CODE REVIEW

**Comment by dangershony** [comment]
You couldn't allow 1 night of 0 PRs?

**Comment by Aprogiena** [comment]
i know, sorry :|

File: Stratis.Bitcoin.Features.BlockStore/BlockStoreLoop.cs [hunk 1]
```diff
@@ -286,9 +286,7 @@ protected virtual void SetHighestPersistedBlock(ChainedBlock block)
         {
             this.logger.LogTrace("({0}:'{1}')", nameof(block), block?.HashBlock);
 
-            var blockRepository = this.BlockRepository as BlockRepository;
-
-            if (blockRepository != null)
+            if (this.BlockRepository is BlockRepository blockRepository)
```
Comment (dangershony) [inline Stratis.Bitcoin.Features.BlockStore/BlockStoreLoop.cs]: This is a really cool new feature of the framework (I assume its new)

File: Stratis.Bitcoin/BlockPulling/BlockPullerBehavior.cs [hunk 2]
```diff
@@ -46,12 +46,12 @@ public class BlockPullerBehavior : NodeBehavior, IBlockPullerBehavior
         /// Token that allows cancellation of async tasks.
         /// It is used during component shutdown.
         /// </summary>
-        public CancellationTokenSource CancellationTokenSource => this.cancellationToken;
+        public CancellationTokenSource CancellationTokenSource { get { return this.cancellationToken; } }
```
Comment (dangershony) [inline Stratis.Bitcoin/BlockPulling/BlockPullerBehavior.cs]: This is odd, we normally allow expression properties

Comment (Aprogiena) [reply Stratis.Bitcoin/BlockPulling/BlockPullerBehavior.cs]: MSVS suggest to do it like this. I mean if we are OK with the `=>` on properties, maybe there is something we can do about the MSVS suggestions so they don't appear.

**Comment by bokobza** [comment]
I'll have a look at this too.

File: Stratis.Bitcoin/BlockPulling/BlockPullerBehavior.cs (hunk 2 above)
Comment (bokobza) [reply Stratis.Bitcoin/BlockPulling/BlockPullerBehavior.cs]: I also prefer the `=>` in this case.

Comment (Aprogiena) [reply Stratis.Bitcoin/BlockPulling/BlockPullerBehavior.cs]: i don't really have a preference except that i like the compiler not to report anything to me, be it error, warning or message

**Comment by Aprogiena** [comment]
so i mean it is perfectly fine for me if you reject this PR
but we should then make MSVS not to make those reports for our projects

**Comment by bokobza** [comment]
yes I'd rather disable the warning in VS.
I'm not rejecting the PR. You can just make this small change and then we can disable these warnings from MSVS.

**Comment by Aprogiena** [comment]
ok, i'll try to find why MSVS suggest those first

**Comment by bokobza** [comment]
I'm gonna push this PR, we can make a rule about which way we want to use, and then either disable the warning or change the code where relevant.

**Comment by Aprogiena** [comment]
can't find any rationale for it, it is called IDE0025

**Pull request merged**